botocore/client.py

```diff
@@ -307,9 +332,27 @@ def __init__(self, serializer, endpoint, response_parser,
         self.meta = ClientMeta(event_emitter, self._client_config,
                                endpoint.host, service_model,
                                self._PY_TO_OP_NAME)
+        self._validate_arguments()
+        self._register_handlers()
+
+    def _validate_arguments(self):
+        if self.meta.s3_addressing_style not in ['virtual', 'default']:
```

Comment:
I think the validation should live elsewhere.  What do you think about the Config owning the validation of the valid values?  It doesn't really make sense to create a Config() object with invalid values, so maybe the Config() object should be validating this when it's created.